redbot/cogs/downloader/downloader.py

```diff
@@ -1056,26 +1056,26 @@ async def _cog_unpin(self, ctx: commands.Context, *cogs: InstalledCog) -> None:
     async def _cog_listpinned(self, ctx: commands.Context):
         """List currently pinned cogs."""
         installed = await self.installed_cogs()
-        pinned_list = sorted([cog.name for cog in installed if cog.pinned], key=str.lower)
+        pinned_list = sorted([cog for cog in installed if cog.pinned], key=lambda cog: cog.name)
```

Comment:
We still want to ignore casing while sorting here:
```suggestion
        pinned_list = sorted([cog for cog in installed if cog.pinned], key=lambda cog: cog.name.lower())
```